src/SketchCanvas.tsx

```diff
@@ -73,25 +73,117 @@ class SketchCanvas extends React.Component<SketchCanvasProps, CanvasState> {
     this._size = {width: 0, height: 0};
     this._initialized = false;
 
-    this.state.text = this._processText(
-      props.text ? props.text.map(t => Object.assign({}, t)) : null,
-    );
-  }
+    this.panResponder = PanResponder.create({
+      // Ask to be the responder:
+      onStartShouldSetPanResponder: (_evt, _gestureState) => true,
+      onStartShouldSetPanResponderCapture: (_evt, _gestureState) => true,
+      onMoveShouldSetPanResponder: (_evt, _gestureState) => true,
+      onMoveShouldSetPanResponderCapture: (_evt, _gestureState) => true,
 
-  UNSAFE_componentWillReceiveProps(nextProps: SketchCanvasProps) {
-    this.setState({
-      text: this._processText(
-        nextProps.text ? nextProps.text.map(t => Object.assign({}, t)) : null,
-      ),
-    });
-  }
+      onPanResponderGrant: (evt, gestureState) => {
+        if (!this.props.touchEnabled) {
+          return;
+        }
+        const e = evt.nativeEvent;
+        this._offset = {x: e.pageX - e.locationX, y: e.pageY - e.locationY};
+        this._path = {
+          id: parseInt(String(Math.random() * 100000000)),
+          color: this.props.strokeColor,
+          width: this.props.strokeWidth,
+          data: [],
+        };
 
-  _processText(text: any) {
-    text &&
-      text.forEach(
-        (t: {fontColor: any}) => (t.fontColor = processColor(t.fontColor)),
-      );
```

Comment:
@jyann upon removal of this, Example 6 does not work and crashes

https://github.com/sourcetoad/react-native-sketch-canvas/assets/54297307/9dd28be0-dac8-48d3-843d-871a16345df5

Reply:
huh, weird, that was working when I tested it last. Good catch, will look into that

Reply:
Added memoized function to address this. Seems to be working as expected, but lmk if I missed anything please

https://github.com/sourcetoad/react-native-sketch-canvas/assets/5407791/297aa0fb-cc94-466d-9e49-79d3fb00e1df